src/opera/template/topology.py

```diff
@@ -68,3 +68,88 @@ def get_artifact(self, params):
         node_name, *rest = params
 
         return self.find_node(node_name).get_artifact(["SELF"] + rest)
+
+    def concat(self, params):
+        if not isinstance(params, list):
+            raise DataError("Concat intrinsic function parameters '{}'"
+                            " should be a list".format(params))
+
+        return self.join([params])
+
+    def join(self, params):
+        if isinstance(params[0], list):
```

Comment:
Now error becomes a little confusing - if the 1 parameter is not a `list `it would return `Too many or too less parameters`. We should probably keep checking the number of parameters

```suggestion
        if 1 <= len(params) <= 2:
            if not isinstance(params[0], list):
                raise DataError("Concat or join intrinsic function"
                                " parameters for '{}'"
                                " should be a list".format(params))
```

Reply:
Yeah, true. I've applied your suggested change.